lib/auth/googleauth.js

```diff
@@ -294,12 +309,25 @@ GoogleAuth.prototype.fromJSON = function(json, opt_callback) {
  */
 GoogleAuth.prototype.fromStream = function(stream, opt_callback) {
   var that = this;
-  var jwtClient = new that.JWTClient();
-  jwtClient.fromStream(stream, function(err) {
-    if (err) {
+  if (!stream) {
+    process.nextTick(function() {
+      callback(
+        opt_callback,
+        new Error('Must pass in a stream containing the service account auth settings.'));
+    });
+    return;
+  }
+  var s = '';
```

Comment:
Also, I can't have `fromStream` delegate to `jwtclient` or `refreshclient` directly because it needs to read the JSON content to determine which to delegate to, and I can't do that before reading the stream and parsing it into JSON, at which point I might as well just delegate in `fromJSON` anyway.